### PR TITLE
Django17

### PR DESCRIPTION
**Pull request by emord**
Should be tested on staging first

cc: @gcapalbo

### CODE REVIEW

**Comment by emord** [comment]
:white_check_mark:

**Comment by gcapalbo** [comment]
@emord What happens when this gets run for the first time - does it have any issues because the tables already exist?

Also, curious to know what kind of testing was done on this - did we make sure the existing tables remain untouched with the same structure and data, and aren't dropped and recreated or anything like that? (I don't think the migrations actually ran on staging because it's still on django 1.6 right?)

cc'ing @czue and @snopoke in case they want to have a look as well

**Comment by gcapalbo** [comment]
Also, we should make sure that we deploy to all environments, especially ones that don't get updated as often (like zambia) before we merge this, in case there are old south migrations that haven't run there yet.

**Comment by emord** [comment]
Since we upgraded to south 1.0.3 it checks south_migrations first. On the upgrade to django 1.7 it will check the names of the tables and if they already exist it will mark the migration as done.

The one downside of doing this now is that if anyone makes a new migration in any of these tables it will need to be done both in migrations and south_migrations, or it could just be done in south and regenerate the initial migration with django 1.7 until django is updated

https://docs.djangoproject.com/en/1.7/topics/migrations/#upgrading-from-south

File: setup.py [hunk 1]
```diff
@@ -21,10 +21,9 @@
         "greenlet",
         "rawes",
         "elasticsearch==0.4.4",
-        'django>=1.3.1,<1.7',
+        'django>=1.3.1,<1.8',
```
Comment (czue) [inline setup.py]: this needs to be stricter, doesn't it?

**Comment by czue** [comment]
@emord is there a clear rollout plan? I agree with @gcapalbo's comments and am generally keen to make sure we have a very explicit migration plan that includes how to deal with something going wrong and making sure it goes out smoothly. maybe you can start a google doc with some details so it's more centralized? i think that one of me, @snopoke or @dannyroberts should sign off on that before we pull the trigger

**Comment by czue** [comment]
sorry I should have read your comment closer before posting. it seems like the plan is pretty clear and hopefully smooth, though I think we still want a place where we can try and pick apart what could go wrong before this goes out.

**Comment by czue** [comment]
also, is there a reason we're not deleting the south migrations? the docs imply we should

File: setup.py (hunk 1 above)
Comment (emord) [inline setup.py]: Yeah it should be django>=1.6 since we're now using atomic

**Comment by emord** [comment]
@czue https://docs.djangoproject.com/en/1.7/topics/migrations/#libraries-third-party-apps

the reason I didn't delete south migrations was that it's a backwards compatible way to support django 1.7 and south at the moment. My plan for deploy was to:

deploy submodules with migrations and south_migrations
deploy django 1.7
after some time delete south_migrations from submodules

That makes it easy if we need to rollback django 1.7 for some reason without having rollback each submodule.

**Comment by czue** [comment]
cool that makes sense

**Comment by gcapalbo** [comment]
Thanks @emord for clarifying, that sounds good.  I think we still want to make sure though that all south migrations get run everywhere (including zambia) before we move to 1.7

**Comment by emord** [comment]
@dannyroberts Tested on django 1.6. Also been running on staging for 2 weeks. Should be good to go.

**Pull request merged**